packages/core/src/Layout/Layout.ts

```diff
@@ -1,14 +1,38 @@
-const { v4: uuidv4 } = require('uuid');
+import { v4 as uuidv4 } from 'uuid';
 
-const { PageSources } = require('../Page/PageSources');
-const { NodeProcessor } = require('../html/NodeProcessor');
+import { PageSources } from '../Page/PageSources';
+import { NodeProcessor, NodeProcessorConfig } from '../html/NodeProcessor';
 
-const logger = require('../utils/logger');
+import { error } from '../utils/logger';
+import VariableProcessor from '../variables/VariableProcessor';
+import { PluginManager } from '../plugins/PluginManager';
+import { SiteLinkManager } from '../html/SiteLinkManager';
 
 const LAYOUT_PAGE_BODY_VARIABLE = 'content';
 
-class Layout {
-  constructor(sourceFilePath, config) {
+export type LayoutConfig = {
+  pluginManager: PluginManager;
+  siteLinkManager: SiteLinkManager;
+  variableProcessor: VariableProcessor;
+  externalManager: any;
```

Comment:
```suggestion
  externalManager: ExternalManager;
```
Would it be better to use the `ExternalManager` class here?